pkg/forwarder/forwarder.go

```diff
@@ -299,7 +299,13 @@ func (f *DefaultForwarder) sendHTTPTransactions(transactions []*HTTPTransaction)
 	}
 
 	for _, t := range transactions {
-		f.highPrio <- t
+		// We don't want to block the collector if the highPrio queue is full
+		select {
+		case f.highPrio <- t:
+		default:
+			log.Errorf("Critical error: the input queue of the forwarder is full: dropping transaction")
```

Comment:
can you remove the `Critical error:` part of the message? I think logging at an error level is enough :)

Do we have a clear idea of the circumstances that would make us hit this code path?  To me this can only happen if, for a sustained time, `number_of_workers/forwarder_timeout < rate_of_payloads_from_aggregator`. Am I missing other cases?

Reply:
Yes that's the only scenario where it could happen, we never experience it anywhere but better safe than sorry.

Reply:
cool 👍  then I think it's worth mentioning this in an inline comment for our future selves :)